test/rematch2.jl

```diff
@@ -25,6 +25,20 @@ end
 
 @testset "@rematch2 tests" begin
 
+@testset "Check that `,if condition end` guards are parsed properly" begin
+    x = true
+    @test (@match2 3 begin
+        ::Int, if x end => 1
```

Comment:
what happens if someone forgets the comma?

Reply:
Syntax error: extra tokens after the end of expression.